### PR TITLE
Immediately fail the task in case of worker pod having a fatal container state

### PR DESCRIPTION
**Pull request by dirrao**
What happened
When the worker pods init/base containers are in a pending state due to fatal container
state reasons, the tasks eventually fail and the pods are deleted. Currently, it has to wait until the worker_pods_pending_timeout even though the worker pods don't recover.

What do you think should happen instead
When the worker pods init/base containers are in a pending state due to fatal container
state reasons, the worker pod doesn't recover. It doesn't make sense to wait until the worker_pods_pending_timeout. Instead, mark the tasks as failed and delete the worker pods.

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/provider.yaml [hunk 1]
```diff
@@ -224,6 +224,16 @@ config:
         type: string
         example: ~
         default: "False"
+      worker_pod_pending_fatal_container_state_reasons:
+        description: |
+          If the worker pods are in a pending state due to a fatal container
+          state reasons, then fail the task and delete the worker pod
+          if delete_worker_pods is True and delete_worker_pods_on_failure is True.
+        version_added: ~
```
Comment (potiuk) [inline airflow/providers/cncf/kubernetes/provider.yaml]: we should add next minor k8s provider version here

**Review by potiuk — APPROVED** [review]
LGTM. small nit

**Comment by potiuk** [comment]
But also would love to get someone else to take a look

File: airflow/providers/cncf/kubernetes/provider.yaml (hunk 1 above)
Comment (dirrao) [reply airflow/providers/cncf/kubernetes/provider.yaml]: version number updated.

**Review by hussein-awala — APPROVED** [review]
This is a nice improvement, I was worried about the sub-reason for each reason (eg: `pull QPS exceeded` which I'm not sure if it's the exact thing in all the K8S versions and distributions), but as the user has the option to update the reasons list, he can fix it without upgrading the provider version if we detect another similar case or any bug.

LGTM

**Pull request merged**